### DISCUSSION TITLE
API-key?

### DISCUSSION THREAD

**Discussion by micro-fx** [Disane87/spoolman-homeassistant#15] [General]
how exactly do you get the API-Key?
I have tried all that i know, and cant seem to figure this one out

actually i input the IP (http://192.168.1.133:7912/), and leave API-Key empty and the error i get is "URL does not return JSON content"

**Comment by Disane87** [comment]
Hey, thanks for using this!

When you get the message `URL does not return JSON content` it seems your spoolman really doesn't return the exppected data. YOu can try to grab your URL and append `api/v1/spool`. Like this (only an example: `https://spoolman.disane.dev/api/v1/spool`)

Now you should get a json with all your spools. In the code I don't do anything else:
https://github.com/Disane87/spoolman-homeassistant/blob/0155d5937b844bb7c4022f5e95a55584ff54a3e2/custom_components/spoolman/coordinator.py#L27

https://github.com/Disane87/spoolman-homeassistant/blob/0155d5937b844bb7c4022f5e95a55584ff54a3e2/custom_components/spoolman/const.py#L9

I will create your error while using you URL to an issue. 


Concerning your problem with the API key you should ask Spoolman (https://github.com/Donkie/Spoolman). I actually implemented the API keys without having it runnign though.

**Reply by micro-fx** [reply]
@Disane87 thank you for the reply. so i added such as http://192.168.1.133:7912/api/v1/spool
and i get some json format, but includes just the first spool info.

so what is the path i should use when adding the HA integration?

if i use the appended url when adding the integration, i get "Failed to connect to the URL. Status code: 404"

for the API-Key i asked, but he has no clue -> https://github.com/Donkie/Spoolman/discussions/131#discussioncomment-7163161

**Reply by Disane87** [reply]
I will remove the api key portion, since it's unused by spoolman. Didn't really know why I implemented this. Sorry!

For further assistance for your json error, I will refer to #17

**Reply by Disane87** [reply]
Closed while removed in latest version
